volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+    # code I do have from it, and will happily share it if anyone else wants to add it.
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="info", component=info.Info, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="psscan", component=psscan.PsScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="thrdscan", component=thrdscan.ThrdScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="handles", component=handles.Handles, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="sessions", component=sessions.Sessions, version=(0, 0, 0)
+            ),
+            requirements.BooleanRequirement(
+                name="identify-expected",
+                description='In the plugin\'s output, replace false with \
+                                                normal where false is the expected result for a Windows machine running normally. \
+                                                Keep in mind that this plugin uses simple checks to identify "normal" behavior, \
+                                                so you may want to double-check the legitimacy of these processes yourself.',
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="physical-offsets",
+                description="List processes with phyiscall offsets instead of virtual offsets.",
+                optional=True,
+            ),
+        ]
+
+    def proc_name_to_string(self, proc):
+        return proc.ImageFileName.cast(
+            "string", max_length=proc.ImageFileName.vol.count, errors="replace"
+        )
+
+    def is_ascii(self, str):
+        return str.split(".")[0].isalnum()
+
+    def filter_garbage_procs(self, proc_list):
+        return [
+            p
+            for p in proc_list
+            if p.is_valid() and self.is_ascii(self.proc_name_to_string(p))
+        ]
+
+    def translate_offset(self, offset):
+        if self.config["physical-offsets"]:
+            return offset
+
+        kernel = self.context.modules[self.config["kernel"]]
+        layer_name = kernel.layer_name
+
+        try:
+            offset = list(
+                self.context.layers[layer_name].mapping(offset=offset, length=0)
+            )[0][2]
```

Comment:
This is a little hard for people reading the code to verify, it would be better if you indexed the first element, but unpacked the second (so `_, _, offset, _, _ = list(self.context.layers[layer_name].mapping(offset=offset, length=0))[0]`)

Reply:
Got it, will revisit this.